### PR TITLE
expression: resize the result for IfXXSig

### LINKED ISSUE #37414: Runtime error: index out of range [0] with length 0
```sql
create table foo(a decimal(65,0));
create table bar(a decimal(65,0), b decimal(65,0));
insert into bar values(0,0),(1,1),(2,2);

insert into foo select if(b>0, if(a/b>1, 1, 2), null) from bar;
ERROR 1105 (HY000): runtime error: index out of range [0] with length 0
```
Version: 6.1.0

### PR DESCRIPTION
**Pull request by XuHuaiyu**
### What problem does this PR solve?

Issue Number: close #37414 

Problem Summary:
The `IfXXSig.fallbackEvalXX` forgets to resize the `result`, thus trigger the index out of range error.

### Check List

Tests 

- [x] Unit test

Side effects

N/A

Documentation

N/A

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by fixdb — COMMENTED** [review]
Will other expressions like `case when` have the same issue?

**Comment by fixdb** [comment]
I am not able to reproduce the issue with `case when` though:
```sql
insert into foo select case when b>0 then  
  case when a/b>1 then  1 else  a/b end 
 else null end 
from bar;
```

I am not sure whether it doesn't have the issue or it just can't expose the issue.

File: expression/integration_test.go [hunk 1]
```diff
@@ -6106,6 +6106,21 @@ func TestRedundantColumnResolve(t *testing.T) {
...
+	defer clean()
+
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists foo")
+	tk.MustExec("drop table if exists bar")
+	tk.MustExec("create table foo(a decimal(65,0));")
+	tk.MustExec("create table bar(a decimal(65,0), b decimal(65,0));")
+	tk.MustExec("insert into bar values(0,0),(1,1),(2,2);")
+	tk.MustExec("insert into foo select if(b>0, if(a/b>1, 1, 2), null) from bar;")
```
Comment (gengliqi) [inline expression/integration_test.go:6120]: How about adding `ifNULL` and `CaseWhen` cases?

**Comment by gengliqi** [comment]
> I am not able to reproduce the issue with `case when` though:
> 
> ```sql
> insert into foo select case when b>0 then  
>   case when a/b>1 then  1 else  a/b end 
>  else null end 
> from bar;
> ```
> 
> I am not sure whether it doesn't have the issue or it just can't expose the issue.

`case when` doesn't have this issue. See line 67 in the below code.
https://github.com/pingcap/tidb/blob/a8f524b31b26cfa4073f2c687bc6ffe55eade37c/expression/generator/control_vec.go#L64-L69

**Comment by XuHuaiyu** [comment]
`IfNull` does not have this problem, neither.
`result` will be resized in line 271.
https://github.com/pingcap/tidb/blob/7425d2f37d919c27edfb685e1bf916eb409c9dd5/expression/generator/control_vec.go#L268-L277

**Review by wshwsh12 — APPROVED** [review]

**Pull request merged**